robotstxt.go

```diff
@@ -208,7 +208,7 @@ func (g *Group) findRule(path string) (ret *rule) {
 				// From Google's spec:
 				// The order of precedence for rules with wildcards is undefined.
 				if l := len(r.pattern.String()); l > prefixLen {
-					prefixLen = len(r.pattern.String())
+					prefixLen = l
```

Comment:
this one looks good, thanks

Reply:
Two nanosecond less, pattern.String() just returning backing expr so no allocations, please look to TrimRightFunc, Replacer and asterisk and to strings.EqualFold.